app/models/media.rb

```diff
@@ -0,0 +1,4 @@
+# The Media resource manages file uploads to the platform
+class Media < ApplicationRecord
+  has_one_attached :upload
```

Comment:
Note: According the rails docs for `has_one_attached`: 
```txt
If the :dependent option isn’t set, the attachment will be purged (i.e. destroyed) whenever the record is destroyed.
```
However we could set the option to perform the attachment purge in a background job, if that becomes preferable:
```ruby
dependent: :purge_later
```

Reply:
Nice, I think background job is slightly preferable from a "reduce ops-between-request-and- response" perspective; but I would be surprised if it was meaningful within the next few years.